app/controllers/sessions_controller.rb

```diff
@@ -2,19 +2,16 @@
 
 class SessionsController < ApplicationController
   before_action :set_session,
-                except: %i[index scheduled unscheduled completed create]
+                except: %i[new index scheduled unscheduled completed]
 
-  def create
-    skip_policy_scope
+  skip_after_action :verify_policy_scoped, only: :new
 
-    @session =
-      Session.create!(
-        academic_year: Date.current.academic_year,
-        team:,
-        programmes: team.programmes
-      )
+  def new
+    location = team.schools.find(params[:location_id])
+
+    session = Session.find_or_create_by!(team:, academic_year:, location:)
```

Comment:
Surprised to see a create in a GET request. I suppose it's ok because it's idempotent?

Reply:
Discussed in the office, decided to ship it like this, it's perhaps not ideal but keeps things simple for the other pages down the line related to sessions, and I will follow up this PR with a unique constraint to avoid race conditions creating multiple sessions for the same school.

Reply:
Index added in https://github.com/nhsuk/manage-vaccinations-in-schools/pull/1846.